include/seqan3/search/fm_index/bi_fm_index_cursor.hpp

```diff
@@ -908,9 +908,24 @@ class bi_fm_index_cursor
                       "The alphabet types of the given text and index differ.");
         assert(index != nullptr);
 
-        size_type const loc = offset() - index->fwd_fm.index[fwd_lb];
-        size_type const query_begin = loc - index->fwd_fm.text_begin_rs.rank(loc + 1) + 1; // Substract delimiters
-        return text | views::join | views::slice(query_begin, query_begin + query_length());
+        // Position of query in concatenated text.
+        size_type const location = offset() - index->fwd_fm.index[fwd_lb];
+
+        // The rank represents the number of delimiters before position `location + 1`.
+        // Furthermore, this number represents the i-th text this (text) position originates from, since each delimiter
+        // corresponds to one text in the whole concatenated text from left-to-right.
```

Comment:
Your comment makes it seem like `index->fwd_fm.text_begin_rs.` stores the position of the delimiters.
But it stores the start position of the texts.
I think this makes it clearer: (Feel free to rephrase it further)
```suggestion
        // The rank represents the number of start positions of individual sequences/texts in the collection 
        // before position `location + 1` and thereby also the number of delimiters.
```